### PR TITLE
update OTN tunnel to align with the latest version of ietf-te

### PR DESCRIPTION
**Pull request by agva123**
Please review the changes.

### CODE REVIEW

**Review by italobusi — CHANGES_REQUESTED** [review]
Few comments:
1. include ietf-te by revision to avoid any future issues when ietf-te is changed
2. put the augmentation to the te label restrictions together (as in topology models)
3. move comments into the description (as in topology models)

See proposed changes in https://github.com/haomianzheng/IETF-ACTN-YANG-Model/pull/85/commits/99c591b3a5631f37be850163fa388fe317fde2d1

**Review by agva123 — COMMENTED** [review]
looks good.

**Review by italobusi — APPROVED** [review]

**Review by haomianzheng — APPROVED** [review]
looks good to go.

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang [hunk 1]
```diff
@@ -493,11 +573,24 @@ module ietf-otn-tunnel {
     }
   }
 
+  /* Augment label-step of the label restrictions for the forwarding
+     direction of path-in-segment of primary reverse path */
+  augment "/te:te/te:tunnels/te:tunnel/"
+        + "te:primary-paths/te:primary-path/"
+        + "te:primary-reverse-path/"
+        + "te:path-in-segment/te:label-restrictions/"
+        + "te:label-restriction/te:label-step/te:technology" {
```
Comment (sergiobelotti) [inline YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:649]: Why no augmentation for label-start and label-end?

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang [hunk 2]
```diff
@@ -532,40 +625,26 @@ module ietf-otn-tunnel {
...
   augment "/te:te/te:tunnels/te:tunnel/"
-        + "te:p2p-primary-paths/te:p2p-primary-path/"
-        + "te:p2p-primary-reverse-path/"
-        + "te:lsps/te:lsp/te:lsp-record-route-information/"
-        + "te:lsp-record-route-information/te:type/te:label/"
-        + "te:label-hop/te:te-label/te:technology" {
+        + "te:primary-paths/te:primary-path/"
+        + "te:primary-reverse-path/"
+        + "te:path-out-segment/te:label-restrictions/"
+        + "te:label-restriction/te:label-step/te:technology" {
```
Comment (sergiobelotti) [inline YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:690]: same comment, why no augmentation for label-start and label end?

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang [hunk 3]
```diff
@@ -661,10 +740,22 @@ module ietf-otn-tunnel {
...
+  /* Augment label-step of the label restrictions for the forwarding
+     direction of path-in-segment of secondary path */
+  augment "/te:te/te:tunnels/te:tunnel/"
+        + "te:secondary-paths/te:secondary-path/"
+        + "te:path-in-segment/te:label-restrictions/"
+        + "te:label-restriction/te:label-step/te:technology" {
+    description "OTN label.";
+    case otn {
+      uses l1-types:otn-label-step;
+    }
```
Comment (sergiobelotti) [inline YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:804]: why no augmentation for label-start and label-end?

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang [hunk 4]
```diff
@@ -697,38 +788,187 @@ module ietf-otn-tunnel {
...
+  /* Augment label-step of the label restrictions for the forwarding
+     direction of path-out-segment of secondary path */
+  augment "/te:te/te:tunnels/te:tunnel/"
+        + "te:secondary-paths/te:secondary-path/"
+        + "te:path-out-segment/te:label-restrictions/"
+        + "te:label-restriction/te:label-step/te:technology" {
+    description "OTN label.";
+    case otn {
+      uses l1-types:otn-label-step;
+    }
```
Comment (sergiobelotti) [inline YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:842]: same comment as above for path-in-segment

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang [hunk 5]
```diff
@@ -697,38 +788,187 @@ module ietf-otn-tunnel {
...
+     path-out-segment of secondary reverse path */
+  augment "/te:te/te:tunnels/te:tunnel/"
+        + "te:secondary-reverse-paths/te:secondary-reverse-path/"
+        + "te:path-out-segment/te:label-restrictions/"
+        + "te:label-restriction/te:label-end/"
+        + "te:te-label/te:technology" {
+    description "OTN label.";
+    case otn {
+      uses l1-types:otn-label-start-end;
+    }
```
Comment (sergiobelotti) [inline YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:979]: Here it is correctly used otn-label-start-end grouping , why not above ?

**Review by sergiobelotti — COMMENTED** [review]
Why no augmentation of label-start and label-end ?

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang (hunk 1 above)
Comment (italobusi) [reply YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:649]: Clarified that the label-start and label-end augmentations are already defined. This PR is adding also the augmentations for the label-step.

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang (hunk 2 above)
Comment (italobusi) [reply YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:690]: see comment above: clarified that the label-start and label-end augmentations are already defined. This PR is adding also the augmentations for the label-step.

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang (hunk 3 above)
Comment (italobusi) [reply YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:804]: see comment above: clarified that the label-start and label-end augmentations are already defined. This PR is adding also the augmentations for the label-step.

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang (hunk 4 above)
Comment (italobusi) [reply YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:842]: see comment above: clarified that the label-start and label-end augmentations are already defined. This PR is adding also the augmentations for the label-step.

File: YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang (hunk 5 above)
Comment (italobusi) [reply YANG/ccamp/otn-tunnel/ietf-otn-tunnel.yang:979]: see comment above: in this case, the PR is adding augmentations for label-start, label-end and label-step.

**Pull request merged**